spec/consensus/gossip/files.md

```diff
@@ -0,0 +1,125 @@
+# Files and Types
+
+Files and types related to the State/Gossip interaction, excluding testing related.
+
+## reactor.go
+- Entry point for the consensus reactor (State+Gossip)
+    - `NewReactor`
+    - `OnStart`
+    - `OnStop`
+    - `SwitchToConsensus`
+
+- Implements the Gossip Layer
+    - Peer management
+        - `InitPeer`
+        - `AddPeer`
+        - `RemovePeer`
+
+    - Message delivery
+        - `Receive`
+            - Processes messages delivered by the Gossip layer and updates consensus (`State`) and peer information (`PeerState`)
+            - Messages are received through multiple channels
+                - `StateChannel`: `NewRoundStepMessage`, `NewValidBlockMessage`, `HasVoteMessage`, `VoteSetMaj23Message`
+                - `DataChannel`: `ProposalMessage`, `ProposalPOLMessage`, `BlockPartMessage`
+                - `VoteChannel`: `VoteMessage`
+                - `VoteBitsChannel`: `VoteSetBitsMessage`
+
+        - `PeerState` update
```

Comment:
The distinction here is:
- `Set*` methods refer to messages produced and processed by the consensus protocol implementation.
- `Apply*` methods refer to "control" messages, exchanged by the consensus reactors of connected peers; they don't have "payload" are not delivered to the consensus protocol implementation, that is why I used "control" for them.